File: frontend/scripts/create-admin-user.js
#!/usr/bin/env node

/**
 * Create Admin User Script
 * 
 * This script creates an initial admin user in Supabase.
 * Run this after setting up your Supabase project.
 * 
 * Usage:
 *   node scripts/create-admin-user.js <email> <password> [name]
 * 
 * Example:
 *   node scripts/create-admin-user.js admin@example.com mypassword123 "Admin User"
 */

import { createClient } from '@supabase/supabase-js';
import { config } from 'dotenv';

// Load environment variables
config({ path: '.env.local' });

const supabaseUrl = process.env.VITE_SUPABASE_URL;
const supabaseServiceRoleKey = process.env.SUPABASE_SERVICE_ROLE_KEY;

if (!supabaseUrl || !supabaseServiceRoleKey) {
  console.error('❌ Missing required environment variables:');
  console.error('   VITE_SUPABASE_URL');
  console.error('   SUPABASE_SERVICE_ROLE_KEY');
  console.error('');
  console.error('Please add these to your .env.local file');
  process.exit(1);
}

// Create Supabase client with service role key for admin operations
const supabase = createClient(supabaseUrl, supabaseServiceRoleKey, {
  auth: {
    autoRefreshToken: false,
    persistSession: false
  }
});

async function createAdminUser(email, password, name = null) {
  try {
    console.log('🔐 Creating admin user...');
    
    const { data, error } = await supabase.auth.admin.createUser({
      email: email,
      password: password,
      user_metadata: {
        name: name,
        role: 'admin'
      },
      email_confirm: true // Skip email confirmation for admin setup
    });

    if (error) {
      throw error;
    }

    console.log('✅ Admin user created successfully!');
    console.log(`   Email: ${email}`);
    console.log(`   User ID: ${data.user.id}`);
    if (name) {
      console.log(`   Name: ${name}`);
    }
    console.log('');
    console.log('🎉 You can now sign in to the application!');
    
  } catch (error) {
    console.error('❌ Failed to create admin user:', error.message);
    process.exit(1);
  }
}

// Parse command line arguments
const args = process.argv.slice(2);

if (args.length < 2) {
  console.log('📋 Create Admin User Script');
  console.log('');
  console.log('Usage:');
  console.log('  node scripts/create-admin-user.js <email> <password> [name]');
  console.log('');
  console.log('Examples:');
  console.log('  node scripts/create-admin-user.js admin@example.com mypassword123');
  console.log('  node scripts/create-admin-user.js admin@example.com mypassword123 "Admin User"');
  console.log('');
  console.log('Requirements:');
  console.log('  - Valid email address');
  console.log('  - Password (minimum 6 characters)');
  console.log('  - Name (optional)');
  console.log('');
  console.log('Environment Variables Required:');
  console.log('  - VITE_SUPABASE_URL');
  console.log('  - SUPABASE_SERVICE_ROLE_KEY');
  process.exit(1);
}

const [email, password, name] = args;

// Basic validation
if (!email.includes('@')) {
  console.error('❌ Invalid email address');
  process.exit(1);
}

if (password.length < 6) {
  console.error('❌ Password must be at least 6 characters long');
  process.exit(1);
}

// Create the admin user
createAdminUser(email, password, name); 

File: backend/create_admin_user.py
#!/usr/bin/env python3
"""
Create Admin User Script

This script creates an initial admin user in Supabase using the service role key.
This should be run from the backend where the service role key is securely stored.

Usage:
    python create_admin_user.py <email> <password> [name]

Example:
    python create_admin_user.py admin@example.com mypassword123 "Admin User"
"""

import sys
import os
import asyncio
from typing import Optional
from dotenv import load_dotenv
from supabase import create_client, Client

# Load environment variables from .env file
load_dotenv()

def get_supabase_admin_client() -> Client:
    """Create Supabase client with service role key for admin operations."""
    supabase_url = os.getenv("SUPABASE_URL")
    supabase_service_key = os.getenv("SUPABASE_SERVICE_KEY")
    
    if not supabase_url or not supabase_service_key:
        print("❌ Missing required environment variables:")
        print("   SUPABASE_URL")
        print("   SUPABASE_SERVICE_KEY") 
        print("")
        print("Please add these to your backend/.env file")
        sys.exit(1)
    
    return create_client(supabase_url, supabase_service_key)

async def create_admin_user(email: str, password: str, name: Optional[str] = None):
    """Create an admin user in Supabase."""
    try:
        print("🔐 Creating admin user...")
        
        supabase = get_supabase_admin_client()
        
        # Create user with admin role
        response = supabase.auth.admin.create_user({
            "email": email,
            "password": password,
            "user_metadata": {
                "name": name or email.split("@")[0],
                "role": "admin"
            },
            "email_confirm": True  # Skip email confirmation for admin setup
        })
        
        if response.user:
            print("✅ Admin user created successfully!")
            print(f"   Email: {email}")
            print(f"   User ID: {response.user.id}")
            if name:
                print(f"   Name: {name}")
            print("")
            print("🎉 You can now sign in to the application!")
        else:
            raise Exception("Failed to create user - no user returned")
            
    except Exception as error:
        print(f"❌ Failed to create admin user: {str(error)}")
        sys.exit(1)

def main():
    """Main function to parse arguments and create admin user."""
    args = sys.argv[1:]
    
    if len(args) < 2:
        print("📋 Create Admin User Script")
        print("")
        print("Usage:")
        print("  python create_admin_user.py <email> <password> [name]")
        print("")
        print("Examples:")
        print("  python create_admin_user.py admin@example.com mypassword123")
        print("  python create_admin_user.py admin@example.com mypassword123 \"Admin User\"")
        print("")
        print("Requirements:")
        print("  - Valid email address")
        print("  - Password (minimum 6 characters)")
        print("  - Name (optional)")
        print("")
        print("Environment Variables Required:")
        print("  - SUPABASE_URL")
        print("  - SUPABASE_SERVICE_KEY")
        sys.exit(1)
    
    email = args[0]
    password = args[1]
    name = args[2] if len(args) > 2 else None
    
    # Basic validation
    if "@" not in email:
        print("❌ Invalid email address")
        sys.exit(1)
    
    if len(password) < 6:
        print("❌ Password must be at least 6 characters long")
        sys.exit(1)
    
    # Create the admin user
    asyncio.run(create_admin_user(email, password, name))

if __name__ == "__main__":
    main() 